src/classes/testAccessController.cls

```diff
@@ -355,4 +1148,81 @@ private class testAccessController {
     	
     	System.assert(arr.size() == max, 'testLimits');
     }
+
+	private static User testReadOnlyProfileUser(boolean insertRecord) {
+		return testReadOnlyProfileUser(false, null, false, null);
+	}
+
+	private static User testReadOnlyProfileUser(boolean insertRecord, ObjectPermissions contactObjectPermission, boolean createReadWriteFieldPermissions, List<FieldPermissions> additonalFieldPerms) {
+		// Built in System profile
+		Profile readOnlyProfileId = [SELECT Id FROM Profile WHERE Name='Read Only'];
+
+		User u = new User(Alias = 'standt', Email='standarduser@esapi.com', 
+            EmailEncodingKey='UTF-8', LastName='Testing', LanguageLocaleKey='en_US', 
+            LocaleSidKey='en_US', ProfileId = readOnlyProfileId.Id, 
+            TimeZoneSidKey='America/Los_Angeles', UserName='standarduser@esapi.com');
+		if(insertRecord) {
+			insert u;
+
+			PermissionSet ps = new PermissionSet(Label = 'mockPs', Name = 'mockPs');
+			insert ps;
+
+			if(contactObjectPermission != null) {
+				contactObjectPermission.ParentId = ps.Id;
+				insert contactObjectPermission;
+			}
+
+			if(createReadWriteFieldPermissions) {
+				List<FieldPermissions> readWritePerms = createFieldPermissions(ps.Id, true, true);
+				if(additonalFieldPerms != null) {
+					for(FieldPermissions fp : additonalFieldPerms) {
+						fp.ParentId = ps.Id;
+						readWritePerms.add(fp);
+					}
+				}
+				
+				insert readWritePerms;
+			}
+
+			// Put the user with the ReadOnly profile into the new Permission Set to get any Object and Field permissions
+			PermissionSetAssignment psa = new PermissionSetAssignment(PermissionSetId = ps.Id, AssigneeId = u.Id);
+			insert psa; 
+		}
+
+		return u;
+	}
+
+	private static List<FieldPermissions> createFieldPermissions(Id permissionSetId, boolean read, boolean edit) {
+
+		string sObjectType = 'Contact';
+
+		List<string> readWriteFields = new List<String>{'Account','Phone','Fax','MobilePhone','HomePhone','OtherPhone','AssistantPhone','Title','Department','AssistantName','LeadSource','Birthdate','Description','Jigsaw','CleanStatus'};
```

Comment:
Another place to swap our jigsaw field

Reply:
As above, removed the Jigsaw and CleanStatus fields from the tests.